src/plugins/advanced_settings/public/header_user_theme_menu.tsx

```diff
@@ -82,6 +93,10 @@ export const HeaderUserThemeMenu = () => {
 
   const onAppearanceSubmit = async (e: SyntheticEvent) => {
     const actions = [setThemeVersion(themeOptions.find((t) => theme === t.value)?.value ?? '')];
+    const result = uiSettings.getWithBrowserSettings<string>('theme:darkMode');
+    const currentDarkMode = enableUserControl
+      ? result.browserValue ?? result.advancedSettingValue ?? result.defaultValue
+      : result.advancedSettingValue ?? result.defaultValue;
```

Comment:
i don't think consumers should be resolving values - think this should come from uisettingsclient